Sources/NIO/ChannelPipeline.swift

```diff
@@ -989,11 +989,13 @@ private extension CloseMode {
 }
 
 /// `Error` that is used by the `ChannelPipeline` to inform the user of an error.
-public enum ChannelPipelineError: Error {
+public enum ChannelPipelineError: Error, Equatable {
```

Comment:
unfortunately because of SemVer we cannot extend this `enum` with extra cases before 2.0.0 is released. Therefore please add a separate type for this error. It could literally just be

```public struct LeftOverBytesError: Error {
    public let leftOverBytes: ByteBuffer
}```

CC @normanmaurer / @Lukasa for input about the name